src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
```

Comment:
The persisters actually have the dependency already https://github.com/Particular/NServiceBus.RavenDB/blob/develop/src/NServiceBus.RavenDB/SagaPersister/RavenDbSagaStorage.cs#L13

So until we redesign the seam with a proper "SagaPersistenceDefinition" similar to the transports we should be good